src/Microsoft.DotNet.Cli.Utils/CommandResolution/CommandResolverArguments.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class CommandResolverArguments
+    {
+        public string CommandName { get; set; }
+
+        public IEnumerable<string> CommandArguments { get; set; }
+
+        public NuGetFramework Framework { get; set; }
+
+        public string ProjectDirectory { get; set; }
+
+        public string Configuration { get; set; }
+
+        public IEnumerable<string> InferredExtensions { get; set; }
+
+        public IEnvironmentProvider Environment { get; set; }
```

Comment:
I had it the way you are suggesting at first, but went back to this because it made it a little easier to test.

I agree though that is a cleaner approach, will make that change back.